docs/_pages/web_client.md

```diff
@@ -220,6 +221,128 @@ web.channels.list((err, res) => {
+const slackAccessTokens = {};
+// Associated team ID
+let teamId = '';
+
+// Fetch access token by team id
+function getTokenByTeamId(id) {
+  if (!slackAccessTokens[id]) {
+    return undefined;
+  } else {
+    return slackAccessTokens[id];
+  }
+}
+
+const accessToken = getTokenByTeamId(teamId);
+
+// Initiate WebClient
+const web = new WebClient(accessToken, {
+  clientId: id,
+  clientSecret: secret,
+  refreshToken: refreshToken
+});
+
+/* When a token is refreshed, a token_refreshed event is triggered:
+ * {
+ *    "access_token": "xoxa-...",
+ *    "expires_in": 86400,
+ *    "team_id": "T12345",
+ *    "enterprise_id": "E1234A12AB"
+ * }
+ */
+web.on('token_refreshed', (event) => {
+  // It's recommended you encrypt and store your new access token in a database to access it later
+  slackAuthorizations[event.team_id] = { access_token: event.access_token, expires_in: event.expires_in };
+  // Save active team ID
+  teamId = event.team_id;
+
+  console.log(`Access token expires in ${event.expires_in}`);
+});
+```
+
+
+---
+
+### Manually handling token rotation
+
+Note: Before implementing it on your own, it's suggested you read through the [token rotation documentation](http://api.slack.com/docs/rotating-and-refreshing-credentials).
+
+If you need more control over token refreshing, you don't need to pass in your refresh token, client ID, or client secret. However, you'll need to listen for `invalid_auth` errors and handle calls to `oauth.access` to fetch new access tokens:
+
+```javascript
+const { WebClient } = require('@slack/client');
+
+const accessToken = process.env.SLACK_TOKEN;
+const refreshToken = process.env.SLACK_REFRESH_TOKEN;
+const id = process.env.SLACK_CLIENT_ID;
+const secret = process.env.SLACK_CLIENT_SECRET;
+
+const web = new WebClient(accessToken);
+
+function refreshToken() {
+  // Custom logic to refresh a token, possibly purging other stale data from db
+  return new Promise((resolve, reject) => {
+    web.oauth.access({
+      client_id: id,
+      client_secret: secret,
+      grant_type: 'refresh_token',
+      refresh_token: refreshToken
+    }).then((res) => {
+      /* Response contains new access token and time until expiration
+       * {
+       *    "access_token": "xoxa-...",
+       *    "expires_in": 86400,
+       *    "team_id": "T12345",
+       *    "enterprise_id": "E1234A12AB"
+       * }
+       */
+      accessToken = res.access_token;
+      // web.token allows you to update the access token for WebClient
+      web.token = accessToken;
+      // Probably purge stale data from db at this point
+
+      resolve();
+    }).catch((err) => {
+      console.log(err);
+      reject();
+    });
+  });
+}
+
+function sendMessage(msg) {
+  return web.chat.postMessage(msg)
+    .then(console.log)
+    .catch((error) => {
+      if (error.code === ErrorCode.PlatformError && error.data.error === 'invalid_auth') {
+        return refreshToken()
```

Comment:
maybe a comment here to say "this error could have occurred because of an expired token, refresh and try again"